addons/api/tests/unit/serializers/credential-store-test.js

```diff
@@ -205,4 +205,42 @@ module('Unit | Serializer | credential store', function (hooks) {
       'client certificate key attribute is expected'
     );
   });
+
+  test('it serializes static-type correctly on create', function (assert) {
+    assert.expect(1);
+    const store = this.owner.lookup('service:store');
+    const record = store.createRecord('credential-store', {
```

Comment:
good idea! But I think this will involve quite a bit of groundwork(find out the duplicated bits in our tests, maybe have a `base` test file with store, API mocks, etc ?) and would go beyond the scope of this pr. For now, I have followed the existing serializer test pattern.

Reply:
@DhariniJeeva 
I mean doing it at test level (within `test)`, not at test suite level (everything within `describe`).

This has to be at test level because every test will have different attributes, right?

My point is, you need to pass and object with attributes to create the record, then, you need to compare that created record with an object. And the object to you need to create the record and the object you need to compare with are identical.

Rather than type twice that object, type it once, reuse it.

This is what I mean:
```
test('it serializes static-type correctly on create', function (assert) {
    assert.expect(1);
    const store = this.owner.lookup('service:store');
    const sampleObject = {
      type: 'static',
      name: 'Static cred store',
      description: 'Description',
    }
    const record = store.createRecord('credential-store', sampleObject);
    assert.deepEqual(record.serialize(), sampleObject);
  });
```

I do not think this is beyond the scope of the PR, but just avoid duplicating code and make it more maintainable. BUT I will not block this PR for this.

Reply:
I'd prefer to leave this as is for the sake of consistency. I think this outweighs saving on a few lines of code. There are other tests in the same file with this format, and I don't want to break the mix and match different patterns in a single file.

Reply:
The reason we don't do this is because there are a number of cases where the input to `createRecord` is different than the expected serialization.  To keep the test formats consistent, I think it's worth the duplication in other cases.

Reply:
Gotcha